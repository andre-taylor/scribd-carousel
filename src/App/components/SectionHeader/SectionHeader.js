import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '../../components';
import { Link } from '../../components';

import styles from './SectionHeader.module.css';

export const SectionHeader = ({ title, subTitle, categoryHref }) => {
  return (
    <header className={styles.sectionHeader}>
      <Typography
        variant="h2"
        className={styles.sectionTitle}
        content={title}
      />
      {categoryHref && (
        <Link
          className={styles.sectionAction}
          content="View More"
          href={categoryHref}
        />
      )}
      {subTitle && (
        <Typography
          variant="p"
          className={styles.sectionSubTitle}
          content={subTitle}
        />
      )}
    </header>
  );
};

SectionHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default SectionHeader;
