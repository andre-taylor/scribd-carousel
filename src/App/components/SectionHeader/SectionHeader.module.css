.sectionHeader {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  padding: 0 6px 18px;
  line-height: 0.75;
  flex-wrap: wrap;
}
.sectionTitle {
  font-size: 2.2rem;
  color: #000514;
}
.sectionAction {
  font-size: 1.6rem;
  color: #a9085e;
}
.sectionSubTitle {
  flex-basis: 100%;
  font-size: 18px;
  font-weight: 700;
  font-style: italic;
  color: #57617a;
  margin: 18px 0 0;
}
