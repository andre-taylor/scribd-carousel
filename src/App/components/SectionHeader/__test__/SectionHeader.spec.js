import React from 'react';
import renderer from 'react-test-renderer';

import SectionHeader from '../SectionHeader';

describe('<SectionHeader />', () => {
  it('renders correctly with all optional and required props', () => {
    const component = renderer.create(
      <SectionHeader
        title={'Buzzy Reads This Week'}
        subTitle={'The books everybodyâ€™s talking about.'}
        categoryHref={
          'https://www.scribd.com/lists/21604319/Buzzy-Reads-This-Week'
        }
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with required props', () => {
    const component = renderer.create(
      <SectionHeader title={'Buzzy Reads This Week'} />
    );
    expect(component).toMatchSnapshot();
  });
});
