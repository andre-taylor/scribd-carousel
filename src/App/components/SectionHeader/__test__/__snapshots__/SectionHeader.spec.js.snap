// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHeader /> renders correctly with all optional and required props 1`] = `
<header
  className="sectionHeader"
>
  <h2
    className="sectionTitle"
  >
    Buzzy Reads This Week
  </h2>
  <a
    className="sectionAction"
    href="https://www.scribd.com/lists/21604319/Buzzy-Reads-This-Week"
  >
    View More
  </a>
  <p
    className="sectionSubTitle"
  >
    The books everybodyâ€™s talking about.
  </p>
</header>
`;

exports[`<SectionHeader /> renders correctly with required props 1`] = `
<header
  className="sectionHeader"
>
  <h2
    className="sectionTitle"
  >
    Buzzy Reads This Week
  </h2>
</header>
`;
